zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java

```diff
@@ -130,27 +121,30 @@ String processedIndexHtml(Resource indexHtml) {
       lensIndex = HttpFileBuilder.of(HttpData.of(processedLensIndexHtml().getBytes(UTF_8)));
     }
 
-    legacyIndex.setHeaders(INDEX_HEADERS);
-    lensIndex.setHeaders(INDEX_HEADERS);
+    ServerCacheControl maxAgeMinute = new ServerCacheControlBuilder().maxAgeSeconds(60).build();
+    legacyIndex.contentType(MediaType.HTML_UTF_8).cacheControl(maxAgeMinute);
+    lensIndex.contentType(MediaType.HTML_UTF_8).cacheControl(maxAgeMinute);
 
     return new IndexSwitchingService(
       legacyIndex.build().asService(), lensIndex.build().asService());
   }
 
   @Bean @Lazy ArmeriaServerConfigurator uiServerConfigurator(
-    IndexSwitchingService indexSwitchingService)
-    throws IOException {
+    IndexSwitchingService indexSwitchingService) throws IOException {
+    ServerCacheControl maxAgeYear =
+      new ServerCacheControlBuilder().maxAgeSeconds(TimeUnit.DAYS.toSeconds(365)).build();
     Service<HttpRequest, HttpResponse> uiFileService =
-      new AddHttpHeadersService(HttpFileService.forClassPath("zipkin-ui")
-        .orElse(HttpFileService.forClassPath("zipkin-lens")), CACHE_YEAR);
-
-
+      HttpFileServiceBuilder.forClassPath("zipkin-ui").cacheControl(maxAgeYear).build()
+        .orElse(HttpFileServiceBuilder.forClassPath("zipkin-lens").cacheControl(maxAgeYear).build());
```

Comment:
Our JS/CSS bundles get a hash appended right?

Reply:
just added a comment that should have been there :)